src/vmshepherd/presets/preset.py

```diff
@@ -83,36 +83,41 @@ def _reset_counters(self):
         """
         if if_older_than is None or time.time() - self._vms_refresh_time > if_older_than:
             self._vms_refresh_time = time.time()
-            self._vms = await self.iaas.list_vms(self.name)
+            self._runtime_vms = await self.iaas.list_vms(self.name)
+        else:
+            runtime_data = await self.runtime_mgr.get_preset_data(self.name)
+            self._vms = runtime_data.iaas['vms']
 
     async def manage(self):
         """ Manage function docstring"""
-
         await self.refresh_vms()
 
-        vms_stat = Counter([vm.get_state() for vm in self._vms])
-        missing = self.count - len(self._vms) if len(self._vms) < self.count else 0
+        vms_stat = Counter([vm.get_state() for vm in self._runtime_vms])
+        missing = self.count - len(self._runtime_vms) if len(self._runtime_vms) < self.count else 0
         logging.info(
             'State: %s, VMs Status: %s expected, %s in iaas, %s running, %s nearby shutdown, %s pending, '
             '%s after time shutdown, %s terminated, %s error, %s unknown, %s missing',
-            'unmanaged' if self.unmanaged else 'managed', self.count, len(self._vms), vms_stat[VmState.RUNNING.value],
-            vms_stat[VmState.NEARBY_SHUTDOWN.value], vms_stat[VmState.PENDING.value],
+            'unmanaged' if self.unmanaged else 'managed', self.count, len(self._runtime_vms),
```

Comment:
What do you think about temporary value for a self._runtime_vms, because this len(self._runtime_vms) is used not once, not twice ;)

Reply:
Fixed, right now is temporary value.